library/game_state.py

```diff
@@ -0,0 +1,167 @@
+        """Main State Machine"""
+        if self.next == 10:
+            self.wait_for_ready(term)
+        elif self.next == 20:
+            self.update_subgrid_select(term)
+        elif self.next == 30:
+            self.update_space_select(term, board)
+        else:
+            pass
+
+    def redraw_user_term(self, term: blessed.Terminal) -> None:
+        """Redraws User Term"""
+        print(term.move_up(7))
+        update_user_section(term, self.term_info)
+
+    def wait_for_ready(self, term: blessed.Terminal) -> None:
+        """Wait for the player to start turn"""
+        self.current = 10
+        if self.player_active == 0:
+            self.player_active = 1
+
+        if self.user_input == "y" or self.user_input == "KEY_ENTER":
+            self.term_info[0] = f"Player {self.player_active} Active"
+
+            if self.user_select_subgrid != 0:
+                self.next = 30  # skip to space select
+                self.term_info[1] = "Select Space by entering 1-9"
+            else:
+                self.next = 20  # go to subgrid select
+                self.term_info[1] = "Select SubGrid by entering 1-9"
+
+            self.term_info[2] = (
+                f"Current: SubGrid {self.user_select_subgrid} "
+                f"| Space {self.user_select_space}"
+            )
+            self.redraw_user_term(term)
+            return None
+
+        else:
+            self.next = 10
+
+        self.redraw_user_term(term)
+
+    def update_subgrid_select(self, term: blessed.Terminal) -> None:
+        """Update the user selected subgrid"""
+        self.current = 20
+
+        if self.user_input in ("1", "2", "3", "4", "5", "6", "7", "8", "9"):
+            self.user_select_subgrid = int(self.user_input)
+            self.term_info[2] = (
+                f"Current: SubGrid {self.user_select_subgrid} "
+                f"| Space {self.user_select_space}"
+            )
+        elif self.user_input == "KEY_ENTER" and self.user_select_subgrid != 0:
+            self.term_info[1] = "Select Space by entering 1-9"
+            self.confirm_entry(term)
+            self.next = 30
+
+        self.redraw_user_term(term)
+
+    def update_space_select(self, term: blessed.Terminal, board: Board) -> None:
+        """Update the user selected space"""
+        self.current = 30
+
+        if self.user_input in ("1", "2", "3", "4", "5", "6", "7", "8", "9"):
+            self.user_select_space = int(self.user_input)
+            self.term_info[2] = (
+                f"Current: SubGrid {self.user_select_subgrid} "
+                f"| Space {self.user_select_space}"
+            )
+        elif self.user_input == "KEY_ENTER" and self.user_select_space != 0:
+            if self.confirm_good_move(board):
+                self.update_board = True  # good to place player token
+                self.confirm_entry(term)
+            else:
+                self.next = 30  # TODO go to error handling and reset values

```

Comment:
Is this TODO something achievable right now?

Reply:
kinda.... I made an attempt. More needs to be done for a proper error handler.

Reply:
I think that's good enough then lol